components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,357 @@
+                                                       handler:self.completionHandler];
+  return action;
+}
+
+@end
+
+@interface MDCActionSheetController () <MDCBottomSheetPresentationControllerDelegate, UITableViewDelegate>
+
+@end
+
+@implementation MDCActionSheetController {
+  MDCActionSheetHeaderView *_header;
+  UITableViewController *_tableViewController;
+  MDCActionSheetDataSource *_dataSource;
+  NSString *_actionSheetTitle;
+  MDCBottomSheetTransitionController *_transitionController;
+  BOOL mdc_adjustFontForContentSizeCategory;
+  /**
+    Used to determine if we need to do a layout within viewWillLayoutSubviews,
+    Because we are setting the preferredContentSize we can no longer get the frame to set the sheet
+    and headers sizes after a transition. 
+   */
+  BOOL needsPreferredContentSizeUpdate;
+}
+
+@synthesize mdc_adjustsFontForContentSizeCategory;
+
++ (instancetype)actionSheetControllerWithTitle:(NSString *)title message:(NSString *)message {
+  return [[MDCActionSheetController alloc] initWithTitle:title message:message];
+}
+
++ (instancetype)actionSheetControllerWithTitle:(NSString *)title {
+  return [MDCActionSheetController actionSheetControllerWithTitle:title message:nil];
+}
+
+- (instancetype)init {
+  return [MDCActionSheetController actionSheetControllerWithTitle:nil message:nil];
+}
+
+- (instancetype)initWithTitle:(NSString *)title message:(NSString *)message {
+  self = [super initWithNibName:nil bundle:nil];
+  if (self) {
+    _actionSheetTitle = [title copy];
+    _message = [message copy];
+    [self commonMDCActionSheetControllerInit];
+  }
+
+  return self;
+}
+
+- (void)commonMDCActionSheetControllerInit {
+  needsPreferredContentSizeUpdate = false;
+  _transitionController = [[MDCBottomSheetTransitionController alloc] init];
+  _transitionController.dismissOnBackgroundTap = YES;
+  /**
+   We have overriden the self implementation so we don't want others to change the values so we
+  have to call super
+   */
+  super.transitioningDelegate = _transitionController;
+  super.modalPresentationStyle = UIModalPresentationCustom;
+  _dataSource = [[MDCActionSheetDataSource alloc] init];
+  _tableViewController = [[UITableViewController alloc] initWithStyle:UITableViewStylePlain];
+  _tableViewController.tableView.delegate = self;
+  _tableViewController.tableView.translatesAutoresizingMaskIntoConstraints = NO;
+  _tableViewController.tableView.estimatedRowHeight = 56.f;
+  _tableViewController.tableView.rowHeight = UITableViewAutomaticDimension;
+  _tableViewController.tableView.separatorStyle = UITableViewCellSeparatorStyleNone;
+  _tableViewController.tableView.scrollEnabled = NO;
+  _tableViewController.tableView.dataSource = _dataSource;
+  [_tableViewController.tableView registerClass:[MDCActionSheetItemTableViewCell class]
+                         forCellReuseIdentifier:kReuseIdentifer];
+  self.backgroundColor = [UIColor whiteColor];
```

Comment:
I think at this point you may be able to remove the UITableViewController usage altogether in favor of just creating and managing a UITableView directly.